### ISSUE TITLE
Validation happens after every epoch, validate_interval is ignored

### ISSUE THREAD

**Issue by villmow** [pytorch/fairseq#2182]
## 🐛 Bug

I used to set the validation (epoch) frequency with `--validate_interval`. Currently on master validation is triggered after every epoch and i can't find a way to trigger validation only every x epochs. 

Has there been a change that caused this behaviour? Is this wanted?

Looking at the code in master, I assume that train.py has been refactored from this:

https://github.com/pytorch/fairseq/blob/9ac98bbd6de414f2acee76a9bc43bf157574b7ec/train.py#L103-L106

To this:

https://github.com/pytorch/fairseq/blob/9f25ffb02ab54a443ba181f732d5dcc00db7bea8/fairseq_cli/train.py#L238-L252

And as we're always at the end of an epoch (L244), the argument `validate_interval` is basically ignored.

### Expected behavior

I should be able to trigger validation only every x epochs, but save epoch_checkpoints after every epoch.

### Environment

 - fairseq Version (e.g., 1.0 or master): master

**Comment by myleott** [comment]
Good point. We generally need a validation loss when saving checkpoints in order to set checkpoint_best.pt properly, but the old behavior kinda ignored this and allowed checkpoint_best to only get updated whenever validation happened to run.

I'll revert this to the old behavior.
